vasppy/kpoints.py

```diff
@@ -41,3 +41,51 @@ def __init__(
             self.shift = np.array([0.0, 0.0, 0.0])
         else:
             self.shift = shift
+
+def get_subdivisions_from_kspacing(kspacing: float, reciprocal_lattice_vectors: np.ndarray) -> tuple[int, int, int]:
+    """
+    Calculate subdivisions along reciprocal lattice vectors from the miniumum allowed distance between k-points (KSPACING).
+
+    Args:
+        kspacing (float): The minimum allowed distance between k-points.
+        reciprocal_lattice_vectors (np.ndarray): The reciprocal lattice vectors. These can be retrieved from ASE as atoms.cell.reciprocal() or from Pymatgen as
+        structure.lattice.reciprocal_lattice_crystallographic.matrix.
+
+    Returns:
+        tuple[int, int, int]: The subdivisions along each reciprocal lattice vector.
+    """
+    subdivisions = []
+    for row in reciprocal_lattice_vectors:
+        magnitude = np.linalg.norm(row)
+        subdivision = np.max([1, np.ceil((magnitude * 2 * np.pi) / kspacing)])
+
+        subdivisions.append(int(subdivision))
+
+    return tuple(subdivisions)
```

Comment:
… example code (not tested)
```suggestion
    def vector_to_subdivision(row: np.ndarray) -> int:
        magnitude = np.linalg.norm(row)
        subdivision = np.max([1, np.ceil((magnitude * 2 * np.pi) / kspacing)])
        return int(subdivision)
    return tuple(map(reciprocal_lattice_vectors))
```

Reply:
Even better, this function is split out entirely for better unit testing
```python
def vector_to_subdivision(k_vector: np.ndarray, 
                          k_spacing: float) -> int:
…
```